wordpress-mobile/WordPress-Android
# [Jetpack Content Migration Flow] Add Help Screen

Fixes #17444

> Depends on #17440 (which also has a dependency).
> The dependency PRs should be reviewed & merged into `trunk` first, afterwards the base branch of this PR should automatically change to `trunk`. At that point the `[Status] Not Ready for Merge` label can be removed, making this PR is ready for review.

This PR implements the Help screen in the WordPress to Jetpack migration flow. This screen is accessible on the Welcome Screen/step via 2 controls:
- The "Need Help?" button
- The profile picture at the top right

The proposal in this PR is to reuse and adapt the existing Help & Support screen that was already in the app, such that we show a different version for this scenario.
This is still open for discussion if we should go forward with it 🙇 .

> 🏁 **All these changes are behind a temporary feature flag.**
>
> 🔔 **Notice:** Once the feature flag is enabled the migration flow will always be presented. The Jetpack app build has to be uninstalled (or "Clear Data" should be done) in order to revert back to the normal functionality. Further development on the data layer will make sure the migration is not attempted again once it's successful. To be partially handled in #17406.

To test:

#### 1️⃣ Test Case 1 - Help in Migration Flow
0. Open the Jetpack app & login
1. Go to `Me` → `App Settings` → `Debug Settings`
2. Enable the `JetpackMigrationFlowFeatureConfig` & restart the app.
3. Tap the `Need Help?` button
4. **Expect** to land on the "Support" screen
5. Navigate back
6. Tap on the Profile Picture at the top right of the screen
7. **Expect** to land on the "Support" screen
8. **Verify** that it has the same elements as shown in the design ([see issue screenshot](https://user-images.githubusercontent.com/4588074/200851387-8af03003-826d-405c-8866-827f2368e027.png))
9. Test the functionality on the support screen
   - **Expect** the buttons to work as they were if support was accessed from the "Me" page in the app.
   - The `WordPress help center` button has the same functionality as the `Browse our FAQ` button on the former support screen.
10. Lastly, tap the `Log out` button
11. **Expect** to land on the app's landing screen screen.

#### 2️⃣ Regression Testing - Former Help & Support Screen
0. Delete the Jetpack app and install it again
1. Open the Jetpack app 
2. Tap `Continue with WordPress.com`
3. Tap the circle with a question mark at the top right
4. **Expect** to see the former Help and Support page of the app ([preview](https://user-images.githubusercontent.com/4588074/201192610-e111433d-59c4-4b79-a89a-0dcdbfa97e2e.png))
5. Go back and continue logging in
6. Go to `Me`
7. Tap `Help & Support`
8. **Expect** to land on the "Help and Support" page, which should be unchanged ([preview](https://user-images.githubusercontent.com/4588074/201192610-e111433d-59c4-4b79-a89a-0dcdbfa97e2e.png))
9. **Verify** that it still works as expected (no regression)
   - There should be no header with the user details
   - There should be no `Log out` button 

## Previews

| Light | Dark |
| - | - |
| <img width="314" src="https://user-images.githubusercontent.com/4588074/201190897-114cb2a0-0f2c-4602-9f5b-5a2bdc6b6d38.png"> | <img width="314" src="https://user-images.githubusercontent.com/4588074/201190892-0e968446-0d80-4926-9fa7-4e2fb1e154f2.png"> |

<details>
<summary>🏞️ <b>Landscape preview</b></summary>

![landscape](https://user-images.githubusercontent.com/4588074/201191134-24904b15-99c4-4d61-804b-869825ddac51.png)
</details>

<details>
<summary>🎬 <b>Video Preview</b></summary>

https://user-images.githubusercontent.com/4588074/201191405-36ad623d-21b4-4250-a329-6f8378976232.mp4
</details>

## Regression Notes
11. Potential unintended areas of impact
   Existing **Help & Support screen** in the app.

12. What I did to test those areas of impact (or what existing automated tests I relied on)
   Manual testing.

13. What automated tests I added (or what prevented me from doing so)
   I added a test for the logout functionality on the help screen, in the new `HelpViewModelTest`.

PR submission checklist:
